### DISCUSSION TITLE
Source ip based routing on linux for different groups of users

### DISCUSSION THREAD

**Discussion by YIvkin** [strongswan/strongswan#1598] [Q&A]
Greetings!
 I am trying to make policy based routing work for different client groups. Client groups have ips from different subnets.
But all of my policy routing is simply ignored.
Example:
A strongswan server have 3 interfaces: 1 public and 2 privates. Based on ip of a client the server needs to select which one of private interfaces packets should be routed to. I add a custom routing table:
`sudo ip route add default via 10.200.0.1 table 101`
and set an ip rule for a client's subnet:
`sudo ip rule add from 192.168.1.0/24 table 101`
But when I trying to `sudo ip route get from 192.168.1.2 to 10.200.0.1` 
I got  `RTNETLINK answers: Network is unreachable`
And when ping from the ip 192.168.1.2 all of the packets are routed via server's default route.
What am I doing wrong and may be there is a better solution ?
Thanks in advance for your answers!

**Comment by tobiasbrunner** [comment]
Is there a route to `10.200.0.1`?

**Reply by YIvkin** [reply]
`10.200.0.2 dev wg0 scope link`
I can ping 10.200.0.1 from the strongswan server.

**Reply by YIvkin** [reply]
```
#sudo ip route get to 10.200.0.1
local 10.200.0.1 dev lo table local src 10.200.0.1 uid 0 
    cache <local> 

```

**Reply by tobiasbrunner** [reply]
It's possible that `ip route get` won't work if the `from` address is not actually a local address (if the server does have an IP in `192.168.1.0/24`, you could try that). Not sure why forwarding wouldn't work, but it's definitely possible that the kernel treats decrypted packets in a weird way.

Instead of using a match on the source IP, using marks (`fwmark`) could be an option (SAs can automatically mark decrypted packets via `set_mark_in` option). But no idea if such routing rules would get matched better.

**Reply by YIvkin** [reply]
Thank you for the idea! I'll test it.

**Reply by YIvkin** [reply]
I did not find how I can set fwmark (mark_in) by condition. Is it possible ?

**Reply by YIvkin** [reply]
Found the problem. It was in the wg interface configuration.

**Reply by tobiasbrunner** [reply]
What was the problem with the interface?

**Reply by YIvkin** [reply]
With understanding of actions order on wireguard interface.
